dashboard/lib/certificate_image.rb

```diff
@@ -183,19 +203,26 @@ def self.accelerated_course?(course)
   end
 
   # assume any unrecognized course name is a hoc course
-  def self.hoc_course?(course)
-    course_version = CurriculumHelper.find_matching_course_version(course)
-    return true if course_version&.hoc?
-    return false if accelerated_course?(course)
-    return false if course_version
-    true
+  def self.hoc_course?(course_name)
+    course_type(course_name) == CERTIFICATE_COURSE_TYPES[:HOC]
+  end
+
+  def self.course_type(course_name)
+    return CERTIFICATE_COURSE_TYPES[:ACCELERATED] if accelerated_course?(course_name)
+
+    course_version = CurriculumHelper.find_matching_course_version(course_name)
+    return CERTIFICATE_COURSE_TYPES[:HOC] if course_version&.hoc?
+    return CERTIFICATE_COURSE_TYPES[:PL] if course_version&.pl_course?
+    return CERTIFICATE_COURSE_TYPES[:OTHER] if course_version
+    CERTIFICATE_COURSE_TYPES[:HOC]
   end
 
   def self.prefilled_title_course?(course)
-    certificate_template_for(course) != 'blank_certificate.png'
+    certificate_template_for(course) != 'blank_certificate.png' && certificate_template_for(course) != 'self_paced_pl_certificate.png'
```

Comment:
Nit: `['blank_certificate.png', 'self_paced_pl_certificate.png'].include? certificate_template_for(course)` (not worth a rebuild though if no other changes)

Reply:
I like that! Don't mind a rebuild -- this PR isn't urgent :)